src/routes/NewUser/components/NewUserPage/NewUserPage.js

```diff
@@ -94,12 +95,67 @@ function NewUser() {
     errors,
     register,
     formState: { isValid, dirty },
+    getValues,
+    setValue,
   } = useForm({
     validationSchema: userProfileSchema,
     defaultValues,
   });
   const [userLocationInfo, setUserLocationInfo] = useState(null);
 
+  /**
+   * Sets the value for the given field if it isn't already populated.
+   * @param {string} fieldName name of the field which will be set
+   * @param {string} allFieldValues current value of all fields
+   * @param {object} googleResultComponent address component returned by the Google Maps API
+   * @param {object} addressType type of address component to lookup
+   * @param {string} addressTypeFormat short_name or long_name
+   */
+  function setValueIfNotAlreadySet(
+    fieldName,
+    allFieldValues,
+    googleResultComponent,
+    addressType,
+    addressTypeFormat = 'long_name',
+  ) {
+    if (
+      googleResultComponent.types.indexOf(addressType) !== -1 &&
+      !allFieldValues[fieldName]
+    ) {
+      setValue(fieldName, googleResultComponent[addressTypeFormat]);
+    }
+  }
+
+  function handleSetUserLocationInfo(locationInfo) {
+    setUserLocationInfo(locationInfo);
+    const result = locationInfo.lookedUpAddress;
+    if (result) {
+      const values = getValues();
+      let streetNumber = '';
+      let streetAddress = '';
+      result.address_components.forEach((component) => {
```

Comment:
In the future we may want to switch to using `reduce` to build a single object so we aren't potentially doing multiple state updates for a single address lookup (due to the for loop)

Reply:
Good point, I wasn't thinking about the state updates.  I won't fix this specific scenario for now, but will keep it in mind going forward.